rust-executor/src/agent/mod.rs

```diff
@@ -86,6 +100,205 @@ impl Into<crate::graphql::graphql_types::AgentSignature> for AgentSignature {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone,)]
+pub struct AgentService {
+    pub did: Option<String>,
+    pub did_document: Option<did_key::Document>,
+    pub signing_key_id: Option<String>,
+    file: String,
+    file_profile: String,
+    pub agent: Option<Agent>,
+}
+
+lazy_static! {
+    static ref AGENT_SERVICE: Arc<Mutex<Option<AgentService>>> = Arc::new(Mutex::new(None));
+}
+
+impl AgentService {
+    pub fn new() -> AgentService {
+        let agent_path = format!("{}/ad4m/agent.json", std::env::var("APPS_DATA_PATH").unwrap());
```

Comment:
Hm, I saw that environment variable in one of the other PRs and thought to comment on it but didn't. I also don't really want to block merging this, but I'm a bit concerned about doing this way. If we want to read this config setting from the environment, I think it should happen once in the initialization. But I saw you're setting it there so it can be read here.

Is there reason why we can't pass this config variable as a variable? `instance()` doesn't have to call `new()`. The value inside the `Arc<Mutex<>>` is already an `Option`, so instance could just return `None` and then we could have an initializtion function which gets the path passed in as variable...

This env variable just looks like its asking for trouble..